Installation issue: root fails to build with intel-oneapi-tbb

<!-- Thanks for taking the time to report this build failure. To proceed with the report please:

1. Title the issue "Installation issue: <name-of-the-package>".
2. Provide the information required below.

We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively! -->

When intel-oneapi-tbb is used to provide tbb, the build of root fails.

### Steps to reproduce the issue

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack spec root ^mesa@:20 | grep tbb
root@6.22.08%gcc@8.4.0~aqua+davix~emacs+examples~fftw~fits~fortran+gdml+gminimal~graphviz+gsl~http~ipo~jemalloc+math~memstat+minuit~mlp~mysql+opengl~postgres~pythia6~pythia8+python~qt4~r+roofit~root7+rpath~shadow~spectrum~sqlite~ssl~table+tbb+threads~tmva+unuran~vc+vdt~vmc+x+xml~xrootd build_type=RelWithDebInfo cxxstd=11 patches=22af3471f3fd87c0fe8917bf9c811c6d806de6c8b9867d30a1e3d383a1b929d7 arch=linux-centos7-broadwell
    ^intel-oneapi-tbb@2021.1.1%gcc@8.4.0 arch=linux-centos7-broadwell
```
```console
$ spack install root ^mesa@:20
...
1 error found in build log:
     96     -- Looking for LibXml2
     97     -- Found LibXml2: /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-8.4.0/libxml2-2.9.10-cgls2d3wuimv62r6pua43
            k443bxyhmv5/lib/libxml2.so (found version "2.9.10")
     98     -- Found FTGL: /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-8.4.0/ftgl-2.4.0-7wrzupbjlehdjrktbsaa6u3xtkls
            givc/include
     99     -- Checking for module 'davix>=0.6.4'
     100    --   Found davix, version 0.7.6
     101    -- Looking for TBB
  >> 102    CMake Error at /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-8.4.0/cmake-3.20.0-br6lwxtmpakaucaykabfl2hg46
            v5dfie/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
     103      Could NOT find TBB (missing: TBB_ROOT_DIR TBB_INCLUDE_DIR TBB_LIBRARY)
     104      (Required is at least version "2018")
     105    Call Stack (most recent call first):
     106      /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-broadwell/gcc-8.4.0/cmake-3.20.0-br6lwxtmpakaucaykabfl2hg46v5dfie/share/
            cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
     107      cmake/modules/FindTBB.cmake:291 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
     108      cmake/modules/SearchInstalledSoftware.cmake:16 (_find_package)

```

### Information on your system

<!-- Please include the output of `spack debug report` -->
```console
$ spack debug report
* **Spack:** 0.16.1-1919-f59ac04
* **Python:** 3.6.8
* **Platform:** linux-centos7-broadwell
* **Concretizer:** original
```

<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists. -->
<!-- * [spack-build-out.txt]()
* [spack-build-env.txt]() -->

<!-- Some packages have maintainers who have volunteered to debug build failures. Run `spack maintainers <name-of-the-package>` and @mention them here if they exist. -->
root maintainers: @HadrienG2 @chissg @drbenmorgan @vvolkl 
intel-oneapi-tbb naintainers: @rscohn2 

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/root/package.py
+++ b/var/spack/repos/builtin/packages/root/package.py
@@ -251,6 +251,8 @@ class Root(CMakePackage):
     # See: https://github.com/root-project/root/issues/6933
     conflicts('^intel-tbb@2021.1:', when='@:6.22',
               msg='Please use an older intel-tbb version')
+    conflicts('^intel-oneapi-tbb@2021.1:', when='@:6.22',
+              msg='Please use an older intel-tbb/intel-oneapi-tbb version')
     # depends_on('intel-tbb@:2021.0', when='@:6.22 ^intel-tbb')
     depends_on('unuran',    when='+unuran')
     depends_on('vc',        when='+vc')

